app/api_topologies.go

```diff
@@ -210,12 +224,40 @@ func updateTopologyFilters(t APITopologyDesc, options []APITopologyOptionGroup)
 	return t
 }
 
-// registry is a threadsafe store of the available topologies
-type registry struct {
+// MakeAPITopologyDesc makes an APITopologyDesc
+func MakeAPITopologyDesc(id string, renderer render.Renderer, name string, rank int, options []APITopologyOptionGroup, hideIfEmpty bool) APITopologyDesc {
+	return APITopologyDesc{id: id, renderer: renderer, Name: name, Rank: rank, HideIfEmpty: hideIfEmpty, Options: options}
+}
+
+//MakeAPITopologyDescWithParent makes an APITopologyDesc with a parent string specified
+func MakeAPITopologyDescWithParent(id string, parent string, renderer render.Renderer, name string, rank int, options []APITopologyOptionGroup, hideIfEmpty bool) APITopologyDesc {
+	return APITopologyDesc{id: id, parent: parent, renderer: renderer, Name: name, Rank: rank, HideIfEmpty: hideIfEmpty, Options: options}
+}
+
+//MakeAPITopologyOptionGroup makes an APITopologyOptionGroup
+func MakeAPITopologyOptionGroup(id string, Default string, options []APITopologyOption) APITopologyOptionGroup {
+	return APITopologyOptionGroup{ID: id, Default: Default, Options: options}
+}
+
+// MakeAPITopologyOption provides an external interface to the package for creating an APITopologyOption
+func MakeAPITopologyOption(value string, label string, filterFunc render.FilterFunc, pseudo bool) APITopologyOption {
+	return APITopologyOption{Value: value, Label: label, filter: filterFunc, filterPseudo: pseudo}
+}
+
+// Registry is a threadsafe store of the available topologies
+type Registry struct {
 	sync.RWMutex
 	items map[string]APITopologyDesc
 }
 
+// MakeRegistry returns a new Registry
```

Comment:
Actually, you could initialize the global topology registry calling this as well.